src/main/flight/mixer.c

```diff
@@ -613,11 +625,22 @@ static void calculateThrottleAndCurrentMotorEndpoints(timeUs_t currentTimeUs)
 
         }
 #endif
+
+        // reduce motorRangeMax when battery is full
+        if (systemConfig()->vbatSagCompEnabled) {
```

Comment:
Why are you using `systemConfig()->vbatSagCompEnabled` here, and `currentPidProfile->vbat_sag_compensation > 0` above? If `vbatSagCompEnabled` is on, and `vbat_sag_compensation` is 0 (i.e. the compensation is disabled for this PID profile) this will result in a divisin by 0 and a wedged flight controller.
Can you please at least test all relevant combinations of configuration values before submitting changes?